packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -900,13 +894,12 @@ export class TreeCheckout implements ITreeCheckoutFork {
 	 */
 	private isRemoteChangeEvent(event: SharedTreeBranchChange<SharedTreeChange>): boolean {
 		return (
-			// remote changes are only ever applied to the main branch
+			// Remote changes are only ever applied to the main branch
 			!this.isBranch &&
-			// remote changes are applied to the main branch by rebasing it onto the trunk,
-			// no other rebases are allowed on the main branch so this means any replaces that are not
-			// transaction commits are remote changes
+			// Remote changes are applied to the main branch by rebasing it onto the trunk.
+			// No other rebases are allowed on the main branch, so we can use this to detect remote changes.
 			event.type === "replace" &&
-			getChangeReplaceType(event) !== "transactionCommit"
+			getChangeReplaceType(event) === "rebase"
```

Comment:
This change here is purely cosmetic, as there are only two possible results from `getChangeReplaceType`